server/server_models.py

```diff
@@ -0,0 +1,173 @@
+        return self.value
+
+
+class TimestampGranularity(str, Enum):
+    WORD = "word"
+    SEGMENT = "segment"
+
+
+class WordObject(BaseModel):
+    start: float
+    end: float
+    word: str
+    probability: float
+
+    @classmethod
+    def from_word(cls, word: Word) -> WordObject:
+        return cls(
+            start=word.start,
+            end=word.end,
+            word=word.word,
+            probability=word.probability,
+        )
+
+
+class SegmentObject(BaseModel):
+    id: int
+    seek: int
+    start: float
+    end: float
+    text: str
+    tokens: list[int]
+    temperature: float
+    avg_logprob: float
+    compression_ratio: float
+    no_speech_prob: float
+
+    @classmethod
+    def from_segment(cls, segment: Segment) -> SegmentObject:
+        return cls(
+            id=segment.id,
+            seek=segment.seek,
+            start=segment.start,
+            end=segment.end,
+            text=segment.text,
+            tokens=segment.tokens,
+            temperature=segment.temperature,
+            avg_logprob=segment.avg_logprob,
+            compression_ratio=segment.compression_ratio,
+            no_speech_prob=segment.no_speech_prob,
+        )
+
+
+class TranscriptionJsonResponse(BaseModel):
+    text: str
+
+    @classmethod
+    def from_segments(cls, segments: list[Segment]) -> TranscriptionJsonResponse:
+        return cls(text=segments_text(segments))
+
+
+class TranscriptionVerboseJsonResponse(BaseModel):
+    task: str = "transcribe"
+    language: str
+    duration: float
+    text: str
+    words: list[WordObject]
+    segments: list[SegmentObject]
+
+    @classmethod
+    def from_segment(
+        cls, segment: Segment, transcription_info: TranscriptionInfo
+    ) -> TranscriptionVerboseJsonResponse:
+        return cls(
+            language=transcription_info.language,
+            duration=segment.end - segment.start,
+            text=segment.text,
+            words=(
+                [WordObject.from_word(word) for word in segment.words]
+                if isinstance(segment.words, list)
+                else []
+            ),
+            segments=[SegmentObject.from_segment(segment)],
+        )
+
+    @classmethod
+    def from_segments(
+        cls, segments: list[Segment], transcription_info: TranscriptionInfo
+    ) -> TranscriptionVerboseJsonResponse:
+        return cls(
+            language=transcription_info.language,
+            duration=transcription_info.duration,
+            text=segments_text(segments),
+            segments=[SegmentObject.from_segment(segment) for segment in segments],
+            words=[
+                WordObject.from_word(word) for word in words_from_segments(segments)
+            ],
+        )
+
+
+class CreateTranscriptionRequest(BaseModel):
+    file: bytes = Field(
+        ...,
+        description="The audio file object (not file name) to transcribe, in one of these formats: flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, or webm.",
```

Comment:
Might want to have this non-OpenAI compatible and instead 'any format supported by ffmpeg'.

Not asking for a change in this commit.